pyscf_ipu/experimental/nuclear_gradients.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+from typing import Callable
+import jax.numpy as jnp
+from jax import jit, tree_map, vmap
+from jax.ops import segment_sum
+
+from .basis import Basis
+from .integrals import _overlap_primitives, _kinetic_primitives
+from .orbital import batch_orbitals
+from .primitive import Primitive
+from .types import Float3, Float3xNxN
+
+
+def grad_primitive_integral(
+    primitive_op: Callable, atom_index: int, a: Primitive, b: Primitive
```

Comment:
What's the signature of this Callable?  It's great to point to an example, even from this more generic code to less generic.